### PR TITLE
vstart: clean up usage a bit

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by tchaikov** [comment]
```
Test all --op dump-journal again
	Journal max_size = 104857600
	Journal max_size = 104857600
	Journal max_size = 104857600
	Journal max_size = 104857600
vstarting....DONE
Wait for health_ok...DONE
Verify erasure coded import data
diff: /tmp/tmp.7911: No such file or directory
ERROR:ECobject2 data not imported properly
Traceback (most recent call last):
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/ceph_objectstore_tool.py", line 1988, in <module>
    status = main(sys.argv[1:])
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/ceph_objectstore_tool.py", line 1776, in main
    ERRORS += verify(DATADIR, EC_POOL, EC_NAME, db)
  File "/home/jenkins-build/build/workspace/ceph-pull-requests/src/test/ceph_objectstore_tool.py", line 231, in verify
    getval = check_output(cmd, shell=True, stderr=DEVNULL)
  File "/usr/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
subprocess.CalledProcessError: Command '/home/jenkins-build/build/workspace/ceph-pull-requests/build/bin/rados -p ec_pool -N 'ns1' getxattr ECobject2 key2-1' returned non-zero exit status 1
```

https://jenkins.ceph.com/job/ceph-pull-requests/17584/

retest this please.

**Comment by tchaikov** [comment]
the problem does not go away with the new "make check" run, @dzafman is this a known issue?

**Comment by dzafman** [comment]
@liewegas @tchaikov The ceph_objectstore_tool.py test starts and stop cluster using vstart.  It also used -n option which has been changed.   I would imagine that the other test failures, may be related to the same issue.

**Comment by liewegas** [comment]
retest this please

**Comment by liewegas** [comment]
retest this please

File: src/vstart.sh [hunk 1]
```diff
@@ -60,10 +60,10 @@ export DYLD_LIBRARY_PATH=$CEPH_LIB:$DYLD_LIBRARY_PATH
 [ -z "$CEPH_NUM_MON" ] && CEPH_NUM_MON=3
 [ -z "$CEPH_NUM_OSD" ] && CEPH_NUM_OSD=3
 [ -z "$CEPH_NUM_MDS" ] && CEPH_NUM_MDS=3
-[ -z "$CEPH_NUM_MGR" ] && CEPH_NUM_MGR=0
+[ -z "$CEPH_NUM_MGR" ] && CEPH_NUM_MGR=1
```
Comment (tchaikov) [inline src/vstart.sh]: ceph-mgr is not included in the "tests" target, see https://github.com/ceph/ceph/blob/master/src/test/CMakeLists.txt#L503. if we want to start mgr by default, we need to include it in "tests".

```
Starting mgr.x
/home/jenkins-build/build/workspace/ceph-pull-requests/build/bin/ceph-mgr -i x
2017-02-02 15:26:32.875139 7f8a4081a300 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-02 15:26:32.875297 7f8a4081a300 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-02 15:26:32.969390 7f980ad29700 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-02 15:26:32.992809 7f980ad29700 -1 WARNING: all dangerous and experimental features are enabled.
/home/jenkins-build/build/workspace/ceph-pull-requests/src/vstart.sh: 329: /home/jenkins-build/build/workspace/ceph-pull-requests/src/vstart.sh: /home/jenkins-build/build/workspace/ceph-pull-requests/build/bin/ceph-mgr: not found

        Start  11: test_pidfile.sh
  4/160 Test   #5: cephtool-test-mon.sh ....................***Failed   14.31 sec
```

see https://jenkins.ceph.com/job/ceph-pull-requests/17881/consoleFull#12243436770189bba6-5c6e-4283-b974-1eceae3f5f28

Comment (liewegas) [inline src/vstart.sh]: hrm, i can't find where the 'tests' dependencies are defined...

Comment (tchaikov) [inline src/vstart.sh]: @liewegas https://github.com/ceph/ceph/blob/master/src/test/CMakeLists.txt#L503

**Comment by liewegas** [comment]
retest this please

**Comment by tchaikov** [comment]
```
*** Caught signal (Segmentation fault) **
 in thread 7f1472ffd700 thread_name:rados
 ceph version 12.0.0-791-g0965e2b (0965e2b73a7dae353210aa31d86be1bc7b8209f4)
 1: (()+0x518e7) [0x7f1485c1a8e7]
 2: (()+0x10330) [0x7f147c68c330]
 3: (MonClient::build_authorizer(int) const+0x57) [0x7f147cb7aff7]
 4: (librados::RadosClient::ms_get_authorizer(int, AuthAuthorizer**, bool)+0x17) [0x7f14856f7557]
 5: (AsyncConnection::_process_connection()+0x88f) [0x7f147cc4225f]
 6: (AsyncConnection::process()+0x6d8) [0x7f147cc46fd8]
 7: (EventCenter::process_events(int)+0x301) [0x7f147cc589d1]
 8: (()+0x3c1e7a) [0x7f147cc5be7a]
 9: (()+0xb1a60) [0x7f147c225a60]
 10: (()+0x8184) [0x7f147c684184]
 11: (clone()+0x6d) [0x7f147b98d37d]
2017-02-25 04:15:31.423799 7f1472ffd700 -1 *** Caught signal (Segmentation fault) **
 in thread 7f1472ffd700 thread_name:rados
```

retest this please.

**Comment by liewegas** [comment]
retest this please

**Comment by tchaikov** [comment]
```
  2/161 Test   #3: ceph_objectstore_tool.py ................***Failed   32.78 sec
vstarting....DONE
Wait for health_ok...DONE
Created Replicated pool #1
Created Erasure coded pool #2
Creating 4 objects in replicated pool
2017-02-25 19:07:21.489373 7f72187338c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.489723 7f72187338c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.492738 7f72187338c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.570137 7f436dca48c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.570251 7f436dca48c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.572433 7f436dca48c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.644098 7f70fec2b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.644345 7f70fec2b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.646936 7f70fec2b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.926451 7f39a3f5b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.926718 7f39a3f5b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.928723 7f39a3f5b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.965387 7f1567f8b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.965608 7f1567f8b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:21.966863 7f1567f8b8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:22.013499 7f088d5aa8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:22.033625 7f088d5aa8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:22.035975 7f088d5aa8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:22.088257 7fb6ef92a8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:22.088359 7fb6ef92a8c0 -1 WARNING: all dangerous and experimental features are enabled.
2017-02-25 19:07:22.089852 7fb6ef92a8c0 -1 WARNING: all dangerous and experimental features are enabled.
created pool rep_pool snap snap1
CRITICAL:Rados put command failed with 139
```

https://jenkins.ceph.com/job/ceph-pull-requests/19139/console,

retest this please.

**Comment by liewegas** [comment]
@tchaikov keep seeing this failure from ceph_objectstore_tool.py: http://tracker.ceph.com/issues/19097

I don't think it's related to wip-vstart though?  Can you take a look? Really need to get this merged since mgr is about to be a required component...

**Comment by dzafman** [comment]
@liewegas Just a piece of information.  I pulled current master and ran ctest -R ceph_objectstore_tool.py and it passed.

**Comment by liewegas** [comment]
It passed!  \o/

**Pull request merged**